### PR TITLE
Deploy from search results

### PR DESCRIPTION
**Pull request by huwshimi**
Fixes: #1692.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- search for something
- hover a charm
- there should be a green plus on the right
- click it
- the charm should be added to the canvas
- do the same with a bundle

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7713/

File: jujugui/static/gui/src/app/components/search-results/item/test-item.js [hunk 1]
```diff
@@ -504,4 +587,70 @@ describe('SearchResultsItem', function() {
     assert.equal(iconsClass,
       'charm-logos__column list-block__column three-col');
   });
+
+  it('deploy a charm', function() {
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/search-results/item/test-item.js]: deploys

File: jujugui/static/gui/src/app/components/search-results/item/test-item.js [hunk 2]
```diff
@@ -504,4 +587,70 @@ describe('SearchResultsItem', function() {
...
+          importBundleYAML={sinon.stub()}
+          item={item}
+          key={item.storeId} />);
+    output.props.children[4].props.children.props.action();
+    assert.equal(changeState.callCount, 1);
+    assert.equal(deployService.callCount, 1);
+    assert.deepEqual(deployService.args[0][0], {this: 'model'});
+  });
+
+  it('deploy a bundle', function() {
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/search-results/item/test-item.js]: deploys

File: jujugui/static/gui/src/app/components/search-results/item/item.js [hunk 3]
```diff
@@ -195,6 +201,57 @@ YUI.add('search-results-item', function(Y) {
...
+    },
+
+    /**
+      Deploy the entity.
+
+      @param entity {Object} The entity to deploy.
+      @param evt {Object} The click event.
+    */
+    _handleDeploy: function(entity, evt) {
+      if (entity.type === 'charm') {
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/search-results/item/item.js]: By passing all of this logic into here it's really duplicating a lot of existing complexity. Have you looked at just passing the `deployTarget` method into here? 

https://github.com/juju/juju-gui/blob/develop/jujugui/static/gui/src/app/app.js#L2233

This method is called by DD as well to extract all the complexity away from adding it to the canvas. It requires the charmstore as the second parameter which will make it difficult to pre-bind, so you may want to swap the order of the arguments first.

Comment (huwshimi) [inline jujugui/static/gui/src/app/components/search-results/item/item.js]: Great suggestion, I didn't know about that method. I've updated it, probably needs another QA/review now.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7719/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7720/

**Comment by huwshimi** [comment]
Thanks @frankban and @hatched

**Comment by huwshimi** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**